src/libraries/analytics/analyticsEventTracking.ts

```diff
@@ -532,12 +532,19 @@ const AnalyticsEventTracking = (() => {
 
       if (!additionalData) {
         if (ENABLE_TOAST && !!SEGMENT_WRITE_KEY) SimpleToast.show(event);
-        return client.track(event);
+        return client.track(event, {
+          segmentAnonymousId: client.userInfo.get()?.anonymousId,
+          segmentUserId: client.userInfo.get()?.userId
+        });
       }
 
       if (ENABLE_TOAST && !!SEGMENT_WRITE_KEY)
         SimpleToast.show(`${event} ${JSON.stringify(additionalData || {})}`);
-      return client.track(event, (additionalData || {}) as JsonMap);
+      return client.track(event, {
+        ...(additionalData || {}),
+        segmentAnonymousId: client.userInfo.get()?.anonymousId,
+        segmentUserId: client.userInfo.get()?.userId
+      } as JsonMap);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for ensuring that `segmentAnonymousId` and `segmentUserId` are always included in the event data sent to Segment. However, there's a potential performance issue because `client.userInfo.get()` is called twice. It would be more efficient to call this method once and store the result in a variable.

```diff
-        return client.track(event, {
-          segmentAnonymousId: client.userInfo.get()?.anonymousId,
-          segmentUserId: client.userInfo.get()?.userId
-        });
+        const userInfo = client.userInfo.get();
+        return client.track(event, {
+          segmentAnonymousId: userInfo?.anonymousId,
+          segmentUserId: userInfo?.userId
+        });

...

-        return client.track(event, {
-          ...(additionalData || {}),
-          segmentAnonymousId: client.userInfo.get()?.anonymousId,
-          segmentUserId: client.userInfo.get()?.userId
-        } as JsonMap);
+        const userInfo = client.userInfo.get();
+        return client.track(event, {
+          ...(additionalData || {}),
+          segmentAnonymousId: userInfo?.anonymousId,
+          segmentUserId: userInfo?.userId
+        } as JsonMap);
```
This change will reduce the number of function calls and improve the performance of your code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->